cornflow-server/cornflow/tests/unit/test_log_in.py

```diff
@@ -29,5 +37,326 @@ def setUp(self):
         self.idx = UserModel.query.filter_by(username="testname").first().id
 
     def test_successful_log_in(self):
+        """
+        Test the successful log in
+        """
         super().test_successful_log_in()
         self.assertEqual(self.idx, self.response.json["id"])
+
+
+class TestLogInOpenAuthNoConfig(CustomTestCase):
+    def create_app(self):
+        """
+        Creates and configures a Flask application for testing.
+
+        :returns: A configured Flask application instance
+        :rtype: Flask
+        """
+        app = create_app("testing-oauth")
+        app.config["SERVICE_USER_ALLOW_PASSWORD_LOGIN"] = 0
+        return app
+
+    def setUp(self):
+        log.root.setLevel(current_app.config["LOG_LEVEL"])
+        db.create_all()
+        access_init_command(verbose=False)
+        register_deployed_dags_command_test(verbose=False)
+        self.user_data = {
+            "username": "testname",
+            "email": "test@test.com",
+            "password": "Testpassword1!",
+        }
+
+        test_user = UserModel(data=self.user_data)
+        test_user.save()
+
+        self.user_data.pop("email")
+
+        self.test_user_id = test_user.id
+
+        self.service_data = {
+            "username": "service_user",
+            "email": "service@test.com",
+            "password": "Testpassword1!",
+        }
+
+        service_user = UserModel(data=self.service_data)
+        service_user.save()
+
+        self.service_data.pop("email")
+        self.service_user_id = service_user.id
+
+        self.assign_role(self.service_user_id, SERVICE_ROLE)
+
+    def test_service_user_login(self):
+        """
+        Tests that a service user can not log in with username and password
+        """
+        response = self.client.post(
+            LOGIN_URL,
+            data=json.dumps(self.service_data),
+            headers={"Content-Type": "application/json"},
+        )
+
+        self.assertEqual(400, response.status_code)
+        self.assertEqual(response.json["error"], "Invalid request")
```

Comment:
Should the "Invalid request" be a constant?

Reply:
Could be but I usually do not generalize those cases on tests.